src/projectile.py

```diff
@@ -22,53 +22,100 @@
 
-    sprite: LinkedSprite[Projectile]
+    sprite: LinkedSprite[Ordnance]
     sprite_lists: SpriteLists
-    muzzle_location: Tuple[float, float, float]
-    damage: float
-    speed: float
-    angle_of_motion: float
+    payload_list: list[Ordnance]
+    """
+    List of Ordnance that are spawned when this Ordnance collides with something. This could be explosions, hit indicators, more rockets that go in every direction, etc.
+    """
     sprite_rotation_offset: float
     exists: bool
     """
-    Projectile is visible and can be collided with.  Used for long-lived projectile objects that are repeatedly added to/removed from the world over time, such as laser beams.
+    Ordnance is visible and can be collided with.  Used for long-lived projectile objects that are repeatedly added to/removed from the world over time, such as laser beams.
     """
 
-    explodes: bool
-
     def __init__(
         self,
-        sprite: LinkedSprite[Projectile],
+        sprite: LinkedSprite[Ordnance],
         sprite_lists: SpriteLists,
-        damage: float,
+        payload_list: list[Ordnance],
     ):
         self.sprite = sprite
         sprite.owner = self
         self.sprite_lists = sprite_lists
-        self.damage = damage
+        self.payload_list = payload_list
+        self.sprite_rotation_offset = 0
+        self.exists = False
+
+    @property
+    def location(self) -> Tuple[float, float, float]:
+        return (
+            self.sprite.center_x,
+            self.sprite.center_y,
+            self.sprite.radians - self.sprite_rotation_offset,
+        )
+
+    def append_sprite(self):
+        self.sprite_lists.ordnance.append(self.sprite)
+        self.exists = True
+
+    def remove_sprite(self):
+        self.sprite_lists.ordnance.remove(self.sprite)
         self.exists = False
-        self.speed = 0
-        self.angle_of_motion = 0
 
-    def setup(
+    def activate_payload(self):
+        for payload in self.payload_list:
+            payload.activate(self.location)
+
+    def activate(self, spawn_location: Tuple[float, float, float]):
+        set_sprite_location(self.sprite, spawn_location)
+        self.append_sprite()
+
+    def update(self, delta_time: float):
+        ...
+
+    def on_collision_with_wall(self, walls_touching_projectile: arcade.SpriteList):
+        ...
+
+    def on_collision_with_player(
+        self, delta_time: float, players_touching_projectile: list[LinkedSprite[Player]]
+    ):
+        ...
+
+
+class Projectile(Ordnance):
+    """
+    A Projectile is a type of Ordncance that is created by a weapon with a direction and speed, and then forgotten about by the weapon.
+    When it collides with something, it activates its payload and is removed
+    """
+
+    damage: float
+    speed: float
+    angle_of_motion: float
+
+    def __init__(
         self,
+        sprite: LinkedSprite[Ordnance],
+        sprite_lists: SpriteLists,
+        payload_list: list[Ordnance],
+        damage: float,
```

Comment:
Probably a better way to do polymorphism here!  We may want to update the damage value in a bunch of ways.

Reply:
I'm not sure I understand what you mean, but we can change how damage is allowed to be updated as the needs arise